.github/workflows/ci.yml

```diff
@@ -1,142 +1,195 @@
+
     - name: Acquire libpng
-      run: curl -fLOSs --retry 3 http://olivevideoeditor.org/deps/libpng-win.7z && 7z x libpng-win.7z
-  
+      run: |
+        curl -fLOSs --retry 3 http://olivevideoeditor.org/deps/libpng-win.7z
+        7z x libpng-win.7z
+
     - name: Acquire libtiff
-      run: curl -fLOSs --retry 3 http://olivevideoeditor.org/deps/libtiff-win.7z && 7z x libtiff-win.7z
-  
+      run: |
+        curl -fLOSs --retry 3 http://olivevideoeditor.org/deps/libtiff-win.7z
+        7z x libtiff-win.7z
+
     - name: Acquire turbojpeg
-      run: curl -fLOSs --retry 3 http://olivevideoeditor.org/deps/turbojpeg-win.7z && 7z x turbojpeg-win.7z
-  
+      run: |
+        curl -fLOSs --retry 3 http://olivevideoeditor.org/deps/turbojpeg-win.7z
+        7z x turbojpeg-win.7z
+
     - name: Acquire zlib
-      run: curl -fLOSs --retry 3 http://olivevideoeditor.org/deps/zlib-win.7z && 7z x zlib-win.7z
-  
+      run: |
+        curl -fLOSs --retry 3 http://olivevideoeditor.org/deps/zlib-win.7z
+        7z x zlib-win.7z
+
     - name: Acquire Boost
-      run: curl -fLOSs --retry 3 http://olivevideoeditor.org/deps/boost-win.7z && 7z x boost-win.7z
-      
+      run: |
+        curl -fLOSs --retry 3 http://olivevideoeditor.org/deps/boost-win.7z
+        7z x boost-win.7z
+
     - name: Acquire Crashpad
-      run: curl -fLOSs --retry 3 http://olivevideoeditor.org/deps/crashpad-win.7z && 7z x crashpad-win.7z
+      run: |
+        curl -fLOSs --retry 3 http://olivevideoeditor.org/deps/crashpad-win.7z
+        7z x crashpad-win.7z
 
     - name: Configure CMake
       # Use a bash shell so we can use the same syntax for environment variable
       # access regardless of the host operating system
       shell: bash
-      working-directory: ${{runner.workspace}}/build
-      # Note the current convention is to use the -S and -B options here to specify source 
-      # and build directories, but this is only available with CMake 3.13 and higher.  
+      working-directory: ${{ runner.workspace }}/build
+      # Note the current convention is to use the -S and -B options here to specify source
+      # and build directories, but this is only available with CMake 3.13 and higher.
       # The CMake binaries on the Github Actions machines are (as of this writing) 3.12
-      run: PATH=$GITHUB_WORKSPACE:$PATH cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G "Ninja"
+      run: |
+        PATH=$GITHUB_WORKSPACE:$PATH \
+          cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=${{ matrix.build-type }} -G "Ninja"
 
     - name: Build
-      working-directory: ${{runner.workspace}}/build
+      working-directory: ${{ runner.workspace }}/build
       shell: bash
-      # Execute the build.  You can specify a specific target with "--target <NAME>"
-      run: cmake --build . --config $BUILD_TYPE
+      # Execute the build. You can specify a specific target with "--target <NAME>"
+      run: |
+        cmake --build . --config ${{ matrix.build-type }}
```

Comment:
I doubt that `--config` is necessary here:

> --config <cfg> = For multi-configuration tools, choose <cfg>.

MSVC is multi-config, in which case `-DCMAKE_BUILD_TYPE` doesn't need to be set in the configure phase because it is decided in the build phase with `--config`. But Ninja is single config AFAIK, so only `-DCMAKE_BUILD_TYPE` is needed.

Reply:
I didn't think so either, but that's just what the GitHub Actions template had and I didn't bother to change it

Reply:
CMake actually supports `-G "Ninja Multi-Config"` to generate multiple .ninja files

https://cmake.org/cmake/help/latest/generator/Ninja%20Multi-Config.html

Calling ninja and cmake correctly looks rather convoluted however.